edgelet/doc/devguide.md

```diff
@@ -1,163 +1,264 @@
 
-To install rustup, visit [https://www.rustup.rs](https://www.rustup.rs/) and follow the instructions.
+1. Build the daemon binaries. If you want to make frequent changes to the daemon's source code and run it without installing it, you probably want this option.
 
-To install or update the compiler:
+
+## Building OS packages
+
+### Linux
+
+Linux packages are built using the `edgelet/build/linux/package.sh` script. Set the following environment variables, then invoke the script:
+
+1. `PACKAGE_OS`: This is the OS on which the resulting packages will be installed. It should be one of `centos7`, `debian8`, `debian9`, `debian10`, `ubuntu16.04` or `ubuntu18.04`
+
+1. `PACKAGE_ARCH`: This is the architecture of the OS on which the resulting packages will be installed. It should be one of `amd64`, `arm32v7` or `aarch64`.
+
+For example:
+
+```sh
+git clone --recurse-submodules 'https://github.com/Azure/iotedge'
+cd iotedge/
+
+PACKAGE_OS='debian10' PACKAGE_ARCH='arm32v7' ./edgelet/build/linux/package.sh
 ```
-rustup update stable
+
+The packages are built inside a Docker container, so no build dependencies are installed on the device running the script. However the user running the script does need to have permissions to invoke the `docker` command.
+
+Note that the script must be run on an `amd64` device. The `PACKAGE_ARCH=arm32v7` and `PACKAGE_ARCH=aarch64` builds are done using a cross-compiler.
+
+Once the packages are built, they will be found somewhere under the `edgelet/target/` directory. (The exact path under that directory depends on the combination of `PACKAGE_OS` and `PACKAGE_ARCH`. See `builds/misc/packages.yaml` for the exact paths.)
+
+If you want to run another build for a different combination of `PACKAGE_OS` and `PACKAGE_ARCH`, make sure to clean the repository first with `sudo git clean -xffd` so that artifacts from the previous build don't get reused for the next one.
+
+
+### Windows
+
+See "Building daemon binaries" below.
+
+
+## Building daemon binaries
+
+### Dependencies
+
+The daemon is written in [Rust,](https://www.rust-lang.org/) so you will need an installation of the Rust compiler. It is recommended to use [`rustup`](https://www.rustup.rs/) to install a Rust toolchain. For Linux, some distributions have their own packages for `rustc` and `cargo`, but these might not match the toolchain used by our code and may fail to build it.
+
+After installing `rustup`, or if you already have it installed, install the toolchain that will be used to build the daemon binaries.
+
+```sh
+rustup self update   # Update rustup itself to the latest version
+
+git clone --recurse-submodules 'https://github.com/Azure/iotedge'
+cd iotedge/edgelet/
+
+rustup update   # Install / update the toolchain used to build the daemon binaries. This is controlled by the rust-toolchain file in this directory.
```

Comment:
This line is too wide and user will need to scroll right to read the full comment. Should we re-arrange comment that the user doesn't need to do additional scrolling?

Reply:
Yeah, good point. Done.